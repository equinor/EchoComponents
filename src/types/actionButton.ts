export interface ActionButton {
    title: string;
    onClick: () => void;
    variant: 'contained' | 'outlined' | 'ghost' | 'ghost_icon' | undefined;
}
