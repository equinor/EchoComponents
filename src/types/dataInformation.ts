export interface DataInformation {
    numberOfItems: number;
    label: string;
    ariaLabel?: string;
    onTagInfoClicked: () => void;
    itemType: string;
    itemButtonClassName?: string;
}
