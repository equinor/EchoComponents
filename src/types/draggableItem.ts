export interface DraggableItem {
    id: number;
    element: React.ReactNode;
}
