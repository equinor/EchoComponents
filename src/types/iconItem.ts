export interface IconItem {
    icon: string;
    onClick: () => void;
}
