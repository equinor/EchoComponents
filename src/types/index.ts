export * from './actionButton';
export * from './dataInformation';
export * from './draggableItem';
export * from './expandableRowProps';
export * from './iconItem';
export * from './iconListItem';
export * from './linkCardItem';
