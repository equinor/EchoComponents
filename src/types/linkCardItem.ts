export interface LinkCardItem {
    title: string;
    url: string;
    target?: '_blank' | '_self';
}
