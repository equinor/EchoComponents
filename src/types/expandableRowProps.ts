import { IconItem } from './iconItem';

export interface ExpandableRowProps {
    iconItems: IconItem[];
    expanded: boolean;
    setExpanded: (status: boolean) => void;
}
