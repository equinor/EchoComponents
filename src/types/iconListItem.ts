import { IconItem } from './iconItem';

export interface IconListItem {
    title?: string;
    subTitle?: string;
    icons: IconItem[];
}
