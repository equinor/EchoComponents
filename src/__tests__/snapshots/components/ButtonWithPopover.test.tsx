import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonWithPopover } from '../../..';
import { DataInformation } from '../../../types/dataInformation';

it('renders ButtonWithPopover with more info button correctly', () => {
    const fetchDataToShow = jest.fn();
    const fetchedDataToShow: DataInformation[] = [];

    const tree = renderer
        .create(
            <ButtonWithPopover
                expanded={true}
                onShowMoreClicked={fetchDataToShow}
                isLoading={false}
                fetchedData={fetchedDataToShow}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
