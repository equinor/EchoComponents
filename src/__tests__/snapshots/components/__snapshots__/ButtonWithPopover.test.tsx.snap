// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ButtonWithPopover with more info button correctly 1`] = `
<div
  className="wrapper"
>
  <button
    className="sc-hKgILt lkgudK button"
    disabled={false}
    onClick={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="sc-dlfnbm iOlYhO"
    >
      <svg
        aria-labelledby="eds-more_horizontal-1"
        className="sc-iwyYcG dqcEJR"
        fill="#007079"
        height="24px"
        name={null}
        role="img"
        title="more options"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="eds-more_horizontal-1"
        >
          more options
        </title>
        <path
          className="sc-cxFLnm"
          clipRule="evenodd"
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-8 2c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
          data-testid="eds-icon-path"
          fillRule="evenodd"
          height="24"
          size={null}
          transform={null}
        />
      </svg>
    </span>
  </button>
</div>
`;
